### PR TITLE
Logging

### PR DESCRIPTION
**Pull request by richardssam**
Have the output of the ffmpeg conversion and any test go to separate log files that can be brought up from the web pages.

There are also some refinements to make the log output a little clearer.

### CODE REVIEW

File: enctests/testframework/main.py [hunk 1]
```diff
@@ -340,34 +368,55 @@ def idiff_compare(source_clip, test_ref, testname, comparisontestinfo):
...
+    default_extract_template = "ffmpeg -y -i {newfile} -compression_level 10 -pred mixed -pix_fmt rgb48be  -frames:v 1 {newpngfile}"
     extract_template = comparisontestinfo.get("extracttemplate", default_extract_template)
 
     # Allow a different image to be compared with, useful for 422 or 420 encoding.
     sourcepng = comparisontestinfo.get("compare_image", source_path.as_posix())
 
-    distortedbase, distortedext = os.path.splitext(distorted)
-    distortedpng = os.path.join(os.path.dirname(distorted), distortedbase + ".png")
+    distortedpng = Path(distorted.parent, distorted.stem + ".png")
+    diffpng = Path(distorted.parent, distorted.stem + "-x20diff.png")
```
Comment (apetrynet) [inline enctests/testframework/main.py:378]:
```suggestion
    distortedpng = Path(distorted.parent, distorted.stem).with_suffix(".png")
    diffpng = Path(distorted.parent, distorted.stem + "-x20diff").with_suffix(".png")
```

File: enctests/testframework/main.py [hunk 2]
```diff
@@ -340,34 +368,55 @@ def idiff_compare(source_clip, test_ref, testname, comparisontestinfo):
...
 
-    extractcmd = extract_template.format(newfile=distorted, newpngfile=distortedpng)
-    print(f"About to extract with cmd: {extractcmd}")
-    result = {'success': False,
-              'result': "undefined"
-    }
-    cmdresult = subprocess.call(shlex.split(extractcmd))
+    if not os.path.exists(distorted):
```
Comment (apetrynet) [inline enctests/testframework/main.py:380]:
```suggestion
    if not distorted.exists():
```

File: enctests/testframework/main.py [hunk 3]
```diff
@@ -414,27 +466,29 @@ def assertresults_compare(source_clip, test_ref, testname, comparisontestinfo):
                 print(f"WARNING: Skipping test since there is no between values, in: {test}")
             values = test.get("between")
             resultstatus = result[testvalue] > values[0] and result[testvalue] < values[1]
-
+            print("Pass" if resultstatus else "Fail", " Parameter:", testvalue, " > ", values[0], " and ", testvalue, " < ", values[1], file=log_file_object)
```
Comment (apetrynet) [inline enctests/testframework/main.py:469]:
```suggestion
            print(f"{'Pass' if resultstatus else 'Fail'} Parameter: {testvalue} > { values[0]} and {testvalue} < {values[1]}", file=log_file_object)
```

File: enctests/testframework/main.py [hunk 4]
```diff
@@ -414,27 +466,29 @@ def assertresults_compare(source_clip, test_ref, testname, comparisontestinfo):
...
- 
+
+            print("Pass" if resultstatus else "Fail", " Parameter:", testvalue, " < ", value, file=log_file_object)
         if testname == "stringmatch":
             if "string" not in test:
                 print(f"WARNING: Skipping test since there is no string to match in : {test}")
                 continue
             value = test.get("string")
             resultstatus = result[testvalue] == value
+            print("Pass" if resultstatus else "Fail", " Parameter:", testvalue, " == ", value, file=log_file_object)
```
Comment (apetrynet) [inline enctests/testframework/main.py:491]:
```suggestion
            print(f"{'Pass' if resultstatus else 'Fail'} Parameter: {testvalue} == {value}", file=log_file_object)
```

**Review by apetrynet — COMMENTED** [review]
This is a very useful addition!
I did a quick lookup if we'd be able to use logging objects and pass them directly to stdout of the subproc calls, but it didn't seem like that works. We can perhaps revisit using logging at some point. Easy to combine stream handlers and file handlers.

Anyway, I've added some suggestions mostly for consistency with f-strings and pathlib use.

**Comment by apetrynet** [comment]
There seems to be a merge conflict. You might need to pull in the latest changes from main.

File: enctests/testframework/encoders/ffmpeg_encoder.py [hunk 5]
```diff
@@ -44,9 +50,12 @@ def run_wedges(self) -> dict:
             test_name = f"{self.test_config.get('name')}-{wedge_name}"
             out_file = self.get_output_filename(test_name)
 
+            if os.path.exists(out_file):
+                os.remove(out_file) # Remove it, so if the new run fails to create anything we are not accidently using the old one.
+
             cmd = self.prep_encoding_command(wedge, out_file)
 
-            print('ffmpeg command:', cmd)
+            print("Creating Wedge:", test_name, " from ", self.get_source_path()[0].name)
```
[the remark this answers is not in the document]
Comment (richardssam) [reply enctests/testframework/encoders/ffmpeg_encoder.py]: Should be resolved.

**Review by apetrynet — APPROVED** [review]
This is a very useful addition!
I did a quick lookup if we'd be able to use logging objects and pass them directly to stdout of the subproc calls, but it didn't seem like that works. We can perhaps revisit using logging at some point. Easy to combine stream handlers and file handlers.

Anyway, I've added some suggestions mostly for consistency with f-strings and pathlib use.

**Pull request merged**